api/handlers.go

```diff
@@ -239,35 +266,52 @@ func (a *API) Mount(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// Only perform the TTL refresh if the driver is in unlocked mode.
-	if !volConfig.Unlocked {
-		if err := a.startTTLRefresh(volName); err != nil {
-			a.HTTPError(w, err)
-			return
-		}
-	}
-
+	// so. if EBUSY is returned here, the resulting unmount will unmount an
+	// existing mount. However, this should never happen because of the above
+	// counter check.
+	// I'm leaving this in because it will break tons of tests if it double
+	// mounts something, after the resulting unmount occurs. This seems like a
+	// great way to fix tons of errors in our code before they ever accidentally
+	// reach a user.
 	mc, err := driver.Mount(driverOpts)
 	if err != nil {
-		a.HTTPError(w, errors.MountFailed.Combine(err))
+		a.clearMount(mountState{w, err, ut, driver, driverOpts, volConfig})
 		return
 	}
 
 	a.MountCollection.Add(mc)
 
+	// Only perform the TTL refresh if the driver is in unlocked mode.
+	if !volConfig.Unlocked {
+		if err := a.startTTLRefresh(volName); err != nil {
+			a.RemoveStopChan(volName)
```

Comment:
I don't know if this is true. I need to research it. Let's just leave it alone for now.